src/librustdoc/html/markdown.rs

```diff
@@ -43,156 +43,19 @@ use html::highlight;
 use html::escape::Escape;
 use test;
 
+use pulldown_cmark::{self, Event, Parser, Tag};
+
 /// A unit struct which has the `fmt::Display` trait implemented. When
 /// formatted, this struct will emit the HTML corresponding to the rendered
 /// version of the contained markdown string.
-pub struct Markdown<'a>(pub &'a str);
+// The second parameter is whether we need a shorter version or not.
```

Comment:
I think I'd prefer an `enum MarkdownOutputStyle { Compact, Fancy }` (or some other fitting variant names) instead of `bool`